Mlem/Views/Tabs/Posts/Components/Post/Components/Community List View/Community List View.swift

```diff
@@ -0,0 +1,285 @@
+            self.hasTestCommunities = true
+        }
+    }
+    
+    var body: some View
+    {
+        VStack {
+            ScrollViewReader { scrollProxy in
+                HStack {
+                    List {
+                        
+                        HomepageFeedRowView(account: account, feedType: .subscribed, iconName: "house.circle.fill", iconColor: Color.red, description: "Subscribed communities from all servers")
+                            .id("top") // For "scroll to top" sidebar item
+                        HomepageFeedRowView(account: account, feedType: .local, iconName: "building.2.crop.circle.fill", iconColor: Color.green, description: "Local communities from your server")
+                        HomepageFeedRowView(account: account, feedType: .all, iconName: "cloud.circle.fill", iconColor: Color.blue, description: "All communities that federate with your server")
+                        
+                        ForEach(communitySections
+                                
+                                // Only show letter headers for letters we have in our community list
+                            .filter({ (communitySection) -> Bool in
+                                getSubscriptionsAndFavorites().contains(where: { communitySection.sidebarEntry.contains(community: $0, isSubscribed: subscribedCommunities.contains($0)) })
+                            })
+                                // Only show sections which have labels to show
+                            .filter({ (communitySection) -> Bool in
+                                communitySection.inlineHeaderLabel != nil
+                            })) { communitySection in
+                                
+                                Section(header:
+                                            HStack {
+                                    Text(communitySection.inlineHeaderLabel!)
+                                    Spacer()
+                                }.id(communitySection.viewId)) {
+                                    ForEach(
+                                        getSubscriptionsAndFavorites()
+                                            .filter({ (listedCommunity) -> Bool in
+                                                // Filter down to sidebar entry which wants us
+                                                communitySection.sidebarEntry.contains(community: listedCommunity, isSubscribed: subscribedCommunities.contains(listedCommunity))
+                                            })
+                                    ) { listedCommunity in
+                                        CommuntiyFeedRowView(account: account, community: listedCommunity, subscribed: subscribedCommunities.contains(listedCommunity), communitySubscriptionChanged: self.hydrateCommunityData)
+                                    }
+                                }
+                            }
+                        
+                    }
```

Comment:
Not a blocker to this merge, but ideally the data filtering happens outside of the body--that enforces nice isolation of logic and rendering, and makes both parts more legible

Reply:
if we can resolve conflicts on this, lets get it done and merge

Reply:
Moved it to a utility method for now.  Definitely more cleanup to do.